src/Ens.ts

```diff
@@ -0,0 +1,461 @@
+    if (!returnee) {
+      throw new ResolutionError(ResolutionErrorCode.RecordNotFound, {
+        domain,
+        recordName: key,
+      });
+    }
+    return returnee;
+  }
+
+  async records(
+    domain: string,
+    keys: string[],
+  ): Promise<Record<string, string>> {
+    const values = await Promise.all(
+      keys.map(async (key) => {
+        if (key.startsWith('crypto.')) {
+          const ticker = key.split('.')[1];
+          return await this.addr(domain, ticker);
+        }
+        if (key === 'ipfs.html.value' || key === 'dweb.ipfs.hash') {
+          return await this.getContentHash(domain);
+        }
+        const ensRecordName = this.fromUDRecordNameToENS(key);
+        return await this.getTextRecord(domain, ensRecordName);
+      }),
+    );
+    return constructRecords(keys, values);
+  }
+
+  async reverse(
+    address: string,
+    currencyTicker: string,
+  ): Promise<string | null> {
+    if (currencyTicker != 'ETH') {
+      throw new Error(`Ens doesn't support any currency other than ETH`);
+    }
+
+    if (address.startsWith('0x')) {
+      address = address.substr(2);
+    }
+
+    const reverseAddress = address + '.addr.reverse';
+    const nodeHash = this.namehash(reverseAddress);
+    const resolverAddress = await this.resolver(reverseAddress).catch(
+      (err: ResolutionError) => {
+        if (err.code === ResolutionErrorCode.UnspecifiedResolver) {
+          return null;
+        }
+        throw err;
+      },
+    );
+
+    if (isNullAddress(resolverAddress)) {
+      return null;
+    }
+
+    const resolverContract = new EthereumContract(
+      resolverInterface(resolverAddress, EthCoinIndex),
+      resolverAddress,
+      this.provider,
+    );
+
+    return await this.resolverCallToName(resolverContract, nodeHash);
+  }
+
+  async reverseOf(
+    address: string,
+    location?: UnsLocation,
+  ): Promise<string | null> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
+      methodName: 'reverseOf',
+    });
+  }
+
+  async getTokenUri(tokenId: string): Promise<string> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
+      method: NamingServiceName.ENS,
+      methodName: 'getTokenUri',
+    });
+  }
+
+  async isAvailable(domain: string): Promise<boolean> {
+    return !(await this.isRegistered(domain));
+  }
+
+  async registryAddress(domain: string): Promise<string> {
+    return this.readerContract.address;
+  }
+
+  async isRegistered(domain: string): Promise<boolean> {
+    const address = await this.owner(domain);
+    return !isNullAddress(address);
+  }
+
+  async getDomainFromTokenId(tokenId: string): Promise<string> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
+      method: NamingServiceName.ENS,
+      methodName: 'getDomainFromTokenId',
+    });
+  }
+
+  locations(domains: string[]): Promise<Locations> {
+    throw new ResolutionError(ResolutionErrorCode.UnsupportedMethod, {
```

Comment:
we should also support this, basically getting info of the domains 
https://github.com/unstoppabledomains/resolution-go/pull/46/files#diff-389a2940b8f151bc1600c8907627ad2630eb44d7b9e93656f0d0973d9588055bR195